tests/scene/test_arraymesh.h

```diff
@@ -0,0 +1,326 @@
+	}
+
+	SUBCASE("Get the blend shape arrays from the mesh.") {
+		TypedArray<Array> blend{};
+		mesh->add_surface_from_arrays(Mesh::PRIMITIVE_TRIANGLES, cylinder_array, blend);
+		CHECK(mesh->surface_get_blend_shape_arrays(2) == blend);
+	}
+}
+
+TEST_CASE("[SceneTree][ArrayMesh] Get/Set mesh metadata and actions") {
+	Ref<ArrayMesh> mesh = memnew(ArrayMesh);
+	Ref<CylinderMesh> cylinder = memnew(CylinderMesh);
+	Array cylinder_array{};
+	cylinder_array.resize(Mesh::ARRAY_MAX);
+	cylinder->create_mesh_array(cylinder_array, 3.f, 3.f, 5.f);
+	mesh->add_surface_from_arrays(Mesh::PRIMITIVE_TRIANGLES, cylinder_array);
+
+	Ref<BoxMesh> box = memnew(BoxMesh);
+	Array box_array{};
+	box_array.resize(Mesh::ARRAY_MAX);
+	box->create_mesh_array(box_array, Vector3(2.f, 1.2f, 1.6f));
+	mesh->add_surface_from_arrays(Mesh::PRIMITIVE_TRIANGLES, box_array);
+
+	SUBCASE("Set the shadow mesh.") {
+		Ref<ArrayMesh> shadow = memnew(ArrayMesh);
+		mesh->set_shadow_mesh(shadow);
+		CHECK(mesh->get_shadow_mesh() == shadow);
+	}
+
+	SUBCASE("Set the shadow mesh multiple times.") {
+		Ref<ArrayMesh> shadow = memnew(ArrayMesh);
+		mesh->set_shadow_mesh(shadow);
+		mesh->set_shadow_mesh(shadow);
+		mesh->set_shadow_mesh(shadow);
+		mesh->set_shadow_mesh(shadow);
+		CHECK(mesh->get_shadow_mesh() == shadow);
+	}
+
+	SUBCASE("Set the same shadow mesh on multiple meshes.") {
+		Ref<ArrayMesh> shadow = memnew(ArrayMesh);
+		Ref<ArrayMesh> mesh2 = memnew(ArrayMesh);
+		mesh->set_shadow_mesh(shadow);
+		mesh2->set_shadow_mesh(shadow);
+
+		CHECK(mesh->get_shadow_mesh() == shadow);
+		CHECK(mesh2->get_shadow_mesh() == shadow);
+	}
+
+	SUBCASE("Set the shadow mesh and then change it.") {
+		Ref<ArrayMesh> shadow = memnew(ArrayMesh);
+		mesh->set_shadow_mesh(shadow);
+		CHECK(mesh->get_shadow_mesh() == shadow);
+		Ref<ArrayMesh> shadow2 = memnew(ArrayMesh);
+		mesh->set_shadow_mesh(shadow2);
+		CHECK(mesh->get_shadow_mesh() == shadow2);
+	}
+
+	SUBCASE("Set custom AABB.") {
+		AABB bound{};
+		mesh->set_custom_aabb(bound);
+		CHECK(mesh->get_custom_aabb() == bound);
+	}
+
+	SUBCASE("Set custom AABB multiple times.") {
+		AABB bound{};
+		mesh->set_custom_aabb(bound);
+		mesh->set_custom_aabb(bound);
+		mesh->set_custom_aabb(bound);
+		mesh->set_custom_aabb(bound);
+		CHECK(mesh->get_custom_aabb() == bound);
+	}
+
+	SUBCASE("Set custom AABB then change to another AABB.") {
+		AABB bound{};
+		AABB bound2{};
+		mesh->set_custom_aabb(bound);
+		CHECK(mesh->get_custom_aabb() == bound);
+		mesh->set_custom_aabb(bound2);
+		CHECK(mesh->get_custom_aabb() == bound2);
+	}
+
+	SUBCASE("Clear all surfaces should leave zero count.") {
+		mesh->clear_surfaces();
+		CHECK(mesh->get_surface_count() == 0);
+	}
+
+	SUBCASE("Able to get correct mesh RID.") {
+		RID rid = mesh->get_rid();
+		CHECK(RS::get_singleton()->mesh_get_surface_count(rid) == 2);
+	}
+
+	SUBCASE("Able to get correct mesh RID.") {
+		RID rid = mesh->get_rid();
+		CHECK(RS::get_singleton()->mesh_get_surface_count(rid) == 2);
+	}
```

Comment:
Looks like this case is duplicated. Also, shouldn't it fail? The previous case clears the surfaces and asserts that surface count is 0

Reply:
@clayjohn Thanks for the catch on the duplicate test. It's my understanding that the code at the top of the TEST_CASE is run before each SUBCASE is run, which is why the test succeeds since the mesh is repopulated with surfaces before this test.

Reply:
Ah, of course. Thanks for the clarification